### PR TITLE
Update documentation to refer to Meson not Make in most places

### PR DESCRIPTION
**Pull request by Ericson2314**
# Motivation

This is necessary to make the Meson one the default and preferred one.

# Context

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

File: doc/manual/src/development/testing.md [hunk 1]
```diff
@@ -135,60 +135,52 @@ Functional tests are run during `installCheck` in the `nix` package build, as we
 
 ### Running the whole test suite
 
-The whole test suite can be run with:
+The whole test suite (functional and unti tests) can be run with:
```
Comment (roberth) [inline doc/manual/src/development/testing.md:138]:
```suggestion
The whole test suite (functional and unit tests) can be run with:
```

File: maintainers/format.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+if ! type -p pre-commit &>/dev/null; then
+  echo "make format: pre-commit not found. Please use \`nix develop\`.";
```
Comment (roberth) [inline maintainers/format.sh:4]:
```suggestion
  echo "format.sh: pre-commit not found. Please use \`nix develop\`.";
```

File: maintainers/format.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  exit 1;
+fi;
+if test -z "$_NIX_PRE_COMMIT_HOOKS_CONFIG"; then
+  echo "make format: _NIX_PRE_COMMIT_HOOKS_CONFIG not set. Please use \`nix develop\`.";
```
Comment (roberth) [inline maintainers/format.sh:8]:
```suggestion
  echo "format.sh: _NIX_PRE_COMMIT_HOOKS_CONFIG not set. Please use \`nix develop\`.";
```

**Review by roberth — COMMENTED** [review]
Skimmed and haven't verified the instructions yet.

File: doc/manual/src/development/building.md [hunk 4]
```diff
@@ -34,16 +34,21 @@ $ nix-shell --attr devShells.x86_64-linux.native-clangStdenvPackages
 To build Nix itself in this shell:
 
 ```console
-[nix-shell]$ autoreconfPhase
-[nix-shell]$ ./configure $configureFlags --prefix=$(pwd)/outputs/out
-[nix-shell]$ make -j $NIX_BUILD_CORES
```
Comment (cole-h) [inline doc/manual/src/development/building.md:39]: Does the `ninjaBuildPhase` do anything sufficiently magic in it, or could we mention it, but also include the "manual invocation"? I typically use the Nix devshell like `nix develop .#native-clangStdenvPackages -c fish`, which (maybe obviously) doesn't support these devshell function-type things.

Comment (Ericson2314) [reply doc/manual/src/development/building.md:39]: Nah it does nothing magic, I was doing `ninja -j20 -k0` until now.

File: doc/manual/src/development/building.md [hunk 5]
```diff
@@ -85,16 +90,20 @@ $ nix develop .#native-clangStdenvPackages
 To build Nix itself in this shell:
 
 ```console
-[nix-shell]$ autoreconfPhase
-[nix-shell]$ configurePhase
-[nix-shell]$ make -j $NIX_BUILD_CORES OPTIMIZE=0
+[nix-shell]$ mesonConfigurePhase
+[nix-shell]$ ninjaBuildPhase
```
[the remark this answers is not in the document]
Comment (Ericson2314) [reply doc/manual/src/development/building.md:94]: See https://github.com/NixOS/nixpkgs/blob/a3835930be390cd330302ae64e29bd4bf5b50683/pkgs/by-name/me/meson/setup-hook.sh#L83 what they do is
```bash
configurePhase=mesonConfigurePhase
```
but variables and functions do not live the same namespace in Bash.

Is it the nicest? No, but that's a Nixpkgs problem. One more thing for our long-overdue stdenv overhaul.

**Pull request merged**